pkg/receive/multitsdb.go

```diff
@@ -154,7 +155,37 @@ type tenant struct {
 	exemplarsTSDB *exemplars.TSDB
 	ship          *shipper.Shipper
 
-	mtx *sync.RWMutex
+	mtx  *sync.RWMutex
+	tsdb *tsdb.DB
+
+	// For tests.
+	blocksToDeleteFn func(db *tsdb.DB) tsdb.BlocksToDeleteFunc
+}
+
+func (t *tenant) blocksToDelete(blocks []*tsdb.Block) map[ulid.ULID]struct{} {
+	if t.tsdb == nil {
+		return nil
+	}
+
+	blocksToDeleteFn := t.blocksToDeleteFn
+	if blocksToDeleteFn == nil {
```

Comment:
Is there a use case where this would be nil?

Reply:
:+1: updated the code not to need this check and fixed the race that popped up